pkg/content/content_clean.go

```diff
@@ -0,0 +1,398 @@
+package content
+
+import (
+	"bufio"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	JcrContentFile    = ".content.xml"
+	JcrMixinTypesProp = "jcr:mixinTypes"
+	JcrRootPrefix     = "<jcr:root"
+	PropPattern       = "^\\s*([^ =]+)=\"([^\"]+)\"(.*)$"
+	NamespacePattern  = "^\\w+:(\\w+)=\"[^\"]+\"$"
+	JcrRoot           = "jcr_root"
+)
+
+type Cleaner struct {
+	config *Opts
+}
+
+func NewCleaner(config *Opts) *Cleaner {
+	return &Cleaner{
+		config: config,
+	}
+}
+
+func (c Cleaner) prepare(root string) error {
+	if c.config.ParentsBackupEnabled {
+		return c.doParentsBackup(root)
+	}
+	return nil
+}
+
+func (c Cleaner) BeforeClean(root string) error {
+	if c.config.ParentsBackupEnabled {
+		return c.doRootBackup(root)
+	}
+	return nil
+}
+
+func (c Cleaner) Clean(root string) error {
+	err := c.flattenFiles(root)
+	if err == nil {
+		if c.config.ParentsBackupEnabled {
+			err = c.undoParentsBackup(root)
+		} else {
+			err = c.cleanParents(root)
+		}
+	}
+	if err == nil {
+		err = c.cleanDotContents(root)
```

Comment:
if some of the following `if` will produce an error then further processing should be interrupted (`return` should be used here)